MekHQ/src/mekhq/campaign/autoresolve/acar/handler/StandardUnitAttackHandler.java

```diff
@@ -0,0 +1,231 @@
+        var attackingUnit = attacker.getUnits().get(attack.getUnitNumber());
+        var toHit = AttackToHitData.compileToHit(game(), attack);
+
+        // Start of attack report
+        reporter.reportAttackStart(attacker, attack.getUnitNumber(), target);
+
+        if (toHit.cannotSucceed()) {
+            reporter.reportCannotSucceed(toHit.getDesc());
+        } else {
+            reporter.reportToHitValue(toHit);
+            Roll roll = Compute.rollD6(2);
+            reporter.reportAttackRoll(roll, attacker);
+
+            if (roll.getIntValue() < toHit.getValue()) {
+                reporter.reportAttackMiss();
+            } else {
+                reporter.reportAttackHit();
+                int damage = calculateDamage(attacker, attack, attackingUnit, target);
+
+                int newArmor = Math.max(0, targetUnit.getCurrentArmor() - damage);
+                reporter.reportDamageDealt(targetUnit, damage, newArmor);
+
+                if (newArmor * 2 <= targetUnit.getCurrentArmor()) {
+                    target.setHighStressEpisode();
+                    reporter.reportStressEpisode();
+                }
+
+                targetUnit.setCurrentArmor(newArmor);
+
+                if (target.isCrippled() && newArmor > 0) {
+                    reporter.reportStressEpisode();
+                    target.setHighStressEpisode();
+                    reporter.reportUnitCrippled();
+                }
+
+                if (newArmor == 0) {
+                    // Destroyed
+                    reporter.reportUnitDestroyed();
+                    target.setHighStressEpisode();
+                    countKill(attackingUnit, targetUnit);
+                } else {
+                    // Check for critical hits if armor is now less than half original
+                    if (newArmor * 2 < targetUnit.getArmor()) {
+                        reporter.reportCriticalCheck();
+                        var critRoll = Compute.rollD6(2);
+                        var criticalRollResult = critRoll.getIntValue();
+                        handleCrits(target, targetUnit, criticalRollResult, attackingUnit);
+                    }
+                }
+            }
+        }
+    }
+
+    private int calculateDamage(Formation attacker, StandardUnitAttack attack,
+                                SBFUnit attackingUnit, Formation target) {
+        int damage = 0;
+        if (attack.getManeuverResult().equals(StandardUnitAttack.ManeuverResult.SUCCESS)) {
+            damage += 1;
+        }
+        damage += attackingUnit.getCurrentDamage().getDamage(attack.getRange()).damage;
+
+        damage = Math.max(1, processDamageByEngagementControl(attacker, target, damage));
+        return damage;
+    }
+
+    private void handleCrits(Formation target, SBFUnit targetUnit, int criticalRollResult, SBFUnit attackingUnit) {
+        if (criticalRollResult <= 4) {
```

Comment:
Could we remove this chain of if-else, replacing it with a switch?

If you want the results to fit between x and y, MathUtility.clamp() has you covered.

Not a major issue, purely stylistic, but I find switches a lot easier to read at a glance when doing results tables.